pkg/kubelet/cm/memorymanager/policy_static.go

```diff
@@ -238,6 +238,63 @@ func (p *staticPolicy) RemoveContainer(s state.State, podUID string, containerNa
 	return nil
 }
 
+func (p *staticPolicy) getPodRequestedResources(pod *v1.Pod) (map[v1.ResourceName]uint64, error) {
+	var reqRsrcsByInitCtrs, reqRsrcsByAppCtrs map[v1.ResourceName]uint64
+
+	for _, ctr := range pod.Spec.InitContainers {
+		reqRsrcs, err := getRequestedResources(&ctr)
+
+		if err != nil {
+			return nil, err
+		}
+		for rsrcName, qty := range reqRsrcs {
+			if _, ok := reqRsrcsByInitCtrs[rsrcName]; !ok {
+				reqRsrcsByInitCtrs[rsrcName] = uint64(0)
+			}
+
+			if reqRsrcs[rsrcName] > reqRsrcsByInitCtrs[rsrcName] {
+				reqRsrcsByInitCtrs[rsrcName] = qty
+			}
+		}
+	}
+
+	for _, ctr := range pod.Spec.Containers {
+		reqRsrcs, err := getRequestedResources(&ctr)
+
+		if err != nil {
+			return nil, err
+		}
+		for rsrcName, qty := range reqRsrcs {
+			if _, ok := reqRsrcsByAppCtrs[rsrcName]; !ok {
+				reqRsrcsByAppCtrs[rsrcName] = uint64(0)
+			}
+
+			reqRsrcsByAppCtrs[rsrcName] += qty
+		}
+	}
+
+	for rsrcName := range reqRsrcsByAppCtrs {
+		if reqRsrcsByInitCtrs[rsrcName] > reqRsrcsByAppCtrs[rsrcName] {
+			reqRsrcsByAppCtrs[rsrcName] = reqRsrcsByInitCtrs[rsrcName]
+		}
+	}
+	return reqRsrcsByAppCtrs, nil
+}
+
+func (p *staticPolicy) GetPodTopologyHints(s state.State, pod *v1.Pod) map[string][]topologymanager.TopologyHint {
+	if v1qos.GetPodQOS(pod) != v1.PodQOSGuaranteed {
+		return nil
+	}
+
+	reqRsrcs, err := p.getPodRequestedResources(pod)
+	if err != nil {
+		klog.Error(err.Error())
+		return nil
+	}
+
```

Comment:
@cynepco3hahue This function does not check extraordinary cases, thoroughly examined in ``GetTopologyHints()``. The question is: should it be copied from the function mentioned and looped over all containers or there's a better solution to the problem?

Reply:
IMHO we should do the same short circuit that done under the `GetTopologyHints`, I think we can move the code:
```golang
		if len(containerBlocks) != len(requestedResources) {
			klog.Errorf("[memorymanager] The number of requested resources by the container %s differs from the number of memory blocks", container.Name)
			return nil
		}

		for _, b := range containerBlocks {
			if _, ok := requestedResources[b.Type]; !ok {
				klog.Errorf("[memorymanager] Container %s requested resources do not have resource of type %s", container.Name, b.Type)
				return nil
			}

			if b.Size != requestedResources[b.Type] {
				klog.Errorf("[memorymanager] Memory %s already allocated to (pod %v, container %v) with different number than request: requested: %d, allocated: %d", b.Type, pod.UID, container.Name, requestedResources[b.Type], b.Size)
				return nil
			}

			containerNUMAAffinity, err := bitmask.NewBitMask(b.NUMAAffinity...)
			if err != nil {
				klog.Errorf("[memorymanager] failed to generate NUMA bitmask: %v", err)
				return nil
			}

			klog.Infof("[memorymanager] Regenerating TopologyHints, %s was already allocated to (pod %v, container %v)", b.Type, pod.UID, container.Name)
			hints[string(b.Type)] = append(hints[string(b.Type)], topologymanager.TopologyHint{
				NUMANodeAffinity: containerNUMAAffinity,
				Preferred:        true,
			})
```

under the separate function and re-use it.

Reply:
@cynepco3hahue Thanks a lot, it seems the most reasonable in that case. Just pushed the code.